app/scripts/lib/ppom/ppom-util.ts

```diff
@@ -29,16 +33,25 @@ export async function validateRequestWithPPOM({
   ppomController,
   request,
   securityAlertId,
+  chainId,
 }: {
   ppomController: PPOMController;
   request: JsonRpcRequest;
   securityAlertId: string;
+  chainId: Hex;
 }): Promise<SecurityAlertResponse> {
   try {
-    return await ppomController.usePPOM(async (ppom: PPOM) => {
-      return await usePPOM(request, securityAlertId, ppom);
-    });
-  } catch (error) {
+    const normalizedRequest = normalizePPOMRequest(request);
+
+    const ppomResponse = isSecurityAlertsAPIEnabled()
+      ? await validateWithAPI(ppomController, chainId, normalizedRequest)
+      : await validateWithController(ppomController, normalizedRequest);
+
+    return {
+      ...ppomResponse,
+      securityAlertId,
+    };
+  } catch (error: unknown) {
     return handlePPOMError(error, 'Error validateRequestWithPPOM#usePPOM: ');
```

Comment:
Should we change the message here for clarity, or add separate `try catch` blocks in each validate method?

Reply:
Sure, I changed the message to add more clarity.